---
title: C6318 | Dokumentace Microsoftu
ms.custom: ''
ms.date: 2018-06-30
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6318
helpviewer_keywords:
- C6318
ms.assetid: 3284a83e-bb8e-461c-adcc-cfc66ceea05e
caps.latest.revision: 15
author: corob-msft
ms.author: gewarren
manager: ghogen
ms.openlocfilehash: 8b737a36066861704e27fa2ce37a980d4674518a
ms.sourcegitcommit: 55f7ce2d5d2e458e35c45787f1935b237ee5c9f8
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 08/22/2018
ms.locfileid: "42669077"
---
# <a name="c6318"></a>C6318
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Nejnovější verzi tohoto tématu můžete najít v [C6318](https://docs.microsoft.com/visualstudio/code-quality/c6318).  
  
upozornění C6318: nesprávně definovaný blok __try /\__except: použití konstanty EXCEPTION_CONTINUE_SEARCH nebo jiné konstanty, která vyhodnocen jako nula ve výrazu filtru výjimky. Není spuštěn kód v bloku obsluhy výjimky  
  
 Toto upozornění signalizuje, že pokud dojde k výjimce v chráněném bloku Tato obslužná rutina strukturované výjimky, výjimka nebude zpracován, protože konstanta `EXCECPTION_CONTINUE_SEARCH` se používá ve výrazu filtru výjimky.  
  
 Tento kód je ekvivalentní chráněného blok bez bloku obsluhy výjimky, protože bloku obsluhy není spuštěn.  
  
## <a name="example"></a>Příklad  
 Následující kód vygeneruje toto upozornění:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except (EXCEPTION_CONTINUE_SEARCH) // warning 6318  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Chcete-li opravit toto upozornění, použijte následující kód:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
void f (char *pch)  
{  
  __try   
  {  
    // assignment might fail  
    *pch = 0;  
  }   
  __except( (GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION) ?  
                EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    puts("Access violation");  
  }  
}   
```  
  
## <a name="see-also"></a>Viz také  
 [try-except – příkaz](http://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)



